File: final_dataset_A/generated_code/claude/python/parse_product_catalog_json_detailed-claude.py
import json
from collections import defaultdict


class CatalogParseError(Exception):
    pass


def load_catalog(json_file):
    try:
        with open(json_file, 'r') as f:
            data = json.load(f)
    except FileNotFoundError:
        raise CatalogParseError(f"File not found: {json_file}")
    except json.JSONDecodeError as e:
        raise CatalogParseError(f"Invalid JSON: {e}")

    if 'products' not in data:
        raise CatalogParseError("Missing 'products' key in catalog")

    return data['products']


def extract_product_info(product):
    return {
        "name": product.get("name", "Unknown"),
        "price": product.get("price", 0),
        "category": product.get("category", "Uncategorized"),
        "sku": product.get("sku")
    }


def group_by_category(products):
    grouped = defaultdict(list)
    for product in products:
        info = extract_product_info(product)
        grouped[info["category"]].append(info)
    return dict(grouped)


def display_grouped_catalog(grouped):
    for category, items in grouped.items():
        print(f"\n{category} ({len(items)} items)")
        for item in items:
            print(f"  - {item['name']}: ${item['price']}")


if __name__ == '__main__':
    try:
        products = load_catalog('catalog.json')
        grouped = group_by_category(products)
        display_grouped_catalog(grouped)
    except CatalogParseError as e:
        print(f"Error: {e}")

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/ParseProductCatalogJSONDetailed-claude.java
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.io.File;
import java.io.IOException;
import java.util.*;

public class ParseProductCatalogJSONDetailed {

    static class Product {
        String name;
        double price;
        String category;
        String sku;

        Product(String name, double price, String category, String sku) {
            this.name = name;
            this.price = price;
            this.category = category;
            this.sku = sku;
        }
    }

    public static List<Product> loadCatalog(String filePath) throws IOException {
        ObjectMapper mapper = new ObjectMapper();
        File file = new File(filePath);
        if (!file.exists()) {
            throw new IOException("File not found: " + filePath);
        }

        JsonNode root = mapper.readTree(file);
        JsonNode productsNode = root.get("products");
        if (productsNode == null) {
            throw new IOException("Missing 'products' key in catalog");
        }

        List<Product> products = new ArrayList<>();
        for (JsonNode node : productsNode) {
            String name = node.hasNonNull("name") ? node.get("name").asText() : "Unknown";
            double price = node.hasNonNull("price") ? node.get("price").asDouble() : 0.0;
            String category = node.hasNonNull("category") ? node.get("category").asText() : "Uncategorized";
            String sku = node.hasNonNull("sku") ? node.get("sku").asText() : null;
            products.add(new Product(name, price, category, sku));
        }
        return products;
    }

    public static Map<String, List<Product>> groupByCategory(List<Product> products) {
        Map<String, List<Product>> grouped = new LinkedHashMap<>();
        for (Product p : products) {
            grouped.computeIfAbsent(p.category, k -> new ArrayList<>()).add(p);
        }
        return grouped;
    }

    public static void main(String[] args) throws IOException {
        List<Product> products = loadCatalog("catalog.json");
        Map<String, List<Product>> grouped = groupByCategory(products);

        for (Map.Entry<String, List<Product>> entry : grouped.entrySet()) {
            System.out.println("\n" + entry.getKey() + " (" + entry.getValue().size() + " items)");
            for (Product p : entry.getValue()) {
                System.out.println("  - " + p.name + ": $" + p.price);
            }
        }
    }
}
